File: backend/src/Hrms.Domain/Entities/EmployeeType.cs
namespace Hrms.Domain.Entities;

/// <summary>
/// Entity สำหรับตาราง 'hrms.employee_type'
/// จัดเก็บประเภทการจ้างงาน (เช่น พนักงานประจำ, พนักงานทดลองงาน, สัญญาจ้าง, รายวัน, พาร์ทไทม์, นักศึกษาฝึกงาน)
/// </summary>
public class EmployeeType
{
    public long Id { get; set; }
    public string TypeCode { get; set; } = string.Empty;
    public string TypeName { get; set; } = string.Empty;
    public string WageType { get; set; } = "MONTHLY"; // MONTHLY, DAILY, HOURLY, STIPEND
    public bool HasSocialSecurity { get; set; } = true;
    public bool HasLeaveEntitlement { get; set; } = true;
    public bool HasOvertime { get; set; } = true;
    public bool HasProvidentFund { get; set; } = false;
    public string Status { get; set; } = "ACTIVE";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public ICollection<EmployeeTypeBenefit> EmployeeTypeBenefits { get; set; } = new List<EmployeeTypeBenefit>();
}

File: frontend/src/types/employeeType.ts
export interface EmployeeType {
  id: number;
  typeCode: string;
  typeName: string;
  wageType: 'MONTHLY' | 'DAILY' | 'HOURLY' | 'STIPEND' | string;
  hasSocialSecurity: boolean;
  hasLeaveEntitlement: boolean;
  hasOvertime: boolean;
  hasProvidentFund: boolean;
  status: 'ACTIVE' | 'INACTIVE';
  createdAt: string;
  updatedAt: string;
  activeContractsCount: number;
}

export interface CreateEmployeeTypePayload {
  typeCode: string;
  typeName: string;
  wageType: string;
  hasSocialSecurity: boolean;
  hasLeaveEntitlement: boolean;
  hasOvertime: boolean;
  hasProvidentFund: boolean;
  status: string;
}

export interface UpdateEmployeeTypePayload {
  typeName: string;
  wageType: string;
  hasSocialSecurity: boolean;
  hasLeaveEntitlement: boolean;
  hasOvertime: boolean;
  hasProvidentFund: boolean;
  status: string;
}

export interface EmployeeTypeStats {
  totalTypes: number;
  monthlyWageCount: number;
  otherWageCount: number;
  activeCount: number;
}
